### DISCUSSION TITLE
How to make examples/angular_bazel_architect work from non-root directory?

### DISCUSSION THREAD

**Discussion by flolu** [bazelbuild/rules_nodejs#2571] [Q&A]
I wan to add the [angular_bazel_architect](https://github.com/bazelbuild/rules_nodejs/tree/stable/examples/angular_bazel_architect) example into a non-root directory.
But trying gives me this error:

```
Workspace configuration file (angular.json, .angular.json, workspace.json, .workspace.json) cannot be found in 'sandbox/linux-sandbox/34/execroot/fullstack-bazel' or in parent directories.
```

So it searches for `angular.json` in the root directory.

How can I tell the `architect-cli` that `angular.json` is located e.g. under `/my-angular-app/angular.json`?
Or is it required for `angular.json` file to be in the root?

**Comment by fischor** [comment]
I think its required to have that in the current working directory (that will be your workspace root). See

https://github.com/angular/angular-cli/blob/5a82689eb48a3eed48fa7620b57a162f0b28c350/packages/angular_devkit/architect_cli/bin/architect.ts#L195-L205

EDIT:
You can use `chdir = package_name()` on your `architect` target definition to change the working directory before execution: https://bazelbuild.github.io/rules_nodejs/Built-ins.html#npm_package_bin-chdir
